league.js

```diff
@@ -0,0 +1,167 @@
+//------------------------------------------------------------------------------
+// Defines a league object which can be used to interact with the spreadsheet
+// for the given league
+//------------------------------------------------------------------------------
+// TODO: create a lichess object that does the same thing for lichess data.
+// TODO: create a set of slack helpers that map channel names to a
+// specific league object, so that the code for scheduling, results or
+// gamelinks 
+var _ = require("underscore");
+var spreadsheets = require("./spreadsheets");
+var moment = require("moment");
+LEAGUE_DEFAULTS = {
+    "spreadsheet": {
+        "key": "",
+        "service_account_auth": {
+            "client_email": "",
+            "private_key": ""
+        },
+        "schedule_colname": ""
+    },
+    "channels": [],
+    "links": {
+        "rules": "",
+        "team": "",
+        "lone-wolf": "",
+        "guide": "",
+        "captains": "",
+        "registration": "",
+        "source": "",
+    }
+};
+
+
+league_attributes = {
+    //--------------------------------------------------------------------------
+    // A list of objects with the following attributes
+    //   - white
+    //   - black
+    //   - scheduled_date (possibly undefined)
+    //   - url (possibly undefined)
+    //   - results (possibly undefined)
+    //--------------------------------------------------------------------------
+    _pairings: [],
+
+    //--------------------------------------------------------------------------
+    // Canonicallized the username
+    //--------------------------------------------------------------------------
+    canonical_username: function(username) {
+        return username.replace("*", "");
+    },
+
+    //--------------------------------------------------------------------------
+    // Syncs the latest team information
+    //--------------------------------------------------------------------------
+    'get_rosters': function(callback) {
+    },
+
+    //--------------------------------------------------------------------------
```

Comment:
In this case, I'm going with "refreshX" for refreshing it from the spreadsheets.  And "findX" for when you're doing a non-exact search and "getX" for when you're asking for _exactly_ this item.

Neither findX or getX necessarily talk to the spreadsheet.  Does that work?